.github/workflows/check-pr.yml

```diff
@@ -0,0 +1,96 @@
+name: check-pr
+
+on:
+  pull_request:
+
+jobs:
+  build-and-test:
+    runs-on: [ ubuntu-22.04 ]
+    steps:
+      - name: checkout v4
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Check out nginx tests
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          repository: nginx/nginx-tests
+          path: nginx-tests
+
+      - name: Install build and test dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install \
+            libedit-dev \
+            libgd-dev \
+            libgd-perl \
+            libgeoip-dev \
+            libpcre2-dev \
+            libperl-dev \
+            libssl-dev \
+            libxml2-dev \
+            libxslt1-dev \
+            zlib1g-dev \
+            ffmpeg \
+            libcache-memcached-perl \
+            libcryptx-perl \
+            libio-socket-ssl-perl \
+            libtest-harness-perl \
+            libprotocol-websocket-perl \
+            libscgi-perl \
+            uwsgi \
+            uwsgi-plugin-python3
```

Comment:
Also, it's worth to consider [caching](https://github.com/actions/cache) all these to speed up regular workflows.

Reply:
You can cache the results of the above installations and reuse it later I guess. Saves some time on (re)downloading + installing, esp. on workflows that run relatively often (I would expect good amount of activity here in future).

Reply:
Again, this is not a blocker, just suggestion. We can track runners' usage to improve things as we go.